src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs

```diff
@@ -422,6 +422,119 @@ public void DestinationFileIsTruncatedWhenItsLargerThanSourceFile()
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory]
+        [InlineData(".leading", "destination")]
+        [InlineData("source", ".leading")]
+        public void CopyWithLeadingDots(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory]
+        [InlineData("-", "destination")]
+        [InlineData("source", "-")]
+        [InlineData("--", "destination")]
+        public void CopyWithDashPrefixedNames(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory]
+        [InlineData("file\tname", "destination")]
+        [InlineData("source", "file\tname")]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void UnixCopyWithEmbeddedControlCharacters(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [ConditionalTheory(nameof(UsingNewNormalization))]
+        [InlineData("trailing ", "destination")]
+        [InlineData("source", "trailing ")]
+        [InlineData("trailing.", "destination")]
+        [InlineData("source", "trailing.")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void WindowsCopyWithTrailingSpacePeriod_ViaExtendedSyntax(string sourceFileName, string destFileName)
+        {
+            // Files with trailing spaces/periods require \\?\ syntax on Windows for creation.
```

Comment:
Implementation of System.IO APIs adds `\\?\` prefix as needed so that users do not need to worry about adding them manually. See:

https://github.com/dotnet/runtime/blob/ce9375b6a1770a6622829876723395b5e9948bc5/src/libraries/Common/src/System/IO/PathInternal.Windows.cs#L94
https://github.com/dotnet/runtime/blob/ce9375b6a1770a6622829876723395b5e9948bc5/src/libraries/Common/src/System/IO/PathInternal.Windows.cs#L109

Does this logic need adjustments?